### PR TITLE
Version 1.1.0

### PR DESCRIPTION
**Pull request by pedropaulofb**
Please evaluate the proposel model for version 1.1.0.

All content was reviewed.

### CODE REVIEW

File: images/Abstract Syntax.png
Comment (claudenirmf) [inline images/Abstract Syntax.png:1]: - What is `{CP1}` in `Project`? The same goes for `Class`
- We need to change the datatype for multilingual text as discussed in the meeting.
- I do not agree that every `Relation` must have one stereotype. We do not have a complete set as of today. Have we discussed this?
- Let's discuss if we want to keep the `propertyType` as optional for `Property`. Maybe we want to raise some message of "incomplete" to the user in cases like this.
- Do not model `CollectiveClass` as a subtype of `Class`. Move the attribute to `Class` and make it optional.
- `Class` must have one or more values for `restrictedTo`

File: images/Concrete Syntax.png
Comment (claudenirmf) [inline images/Concrete Syntax.png:1]: - Just a curiosity: is there any difference between `Text` and `Rectangle` today? I guess there will be when we add support to text formatting.
- Another curiosity: is the association between `Shape` and `ElementView` an aggregation?

File: images/Project.png
Comment (claudenirmf) [inline images/Project.png:1]: - Review diagrams as you make changes (e.g., check if the additional attributes of `OntoumlElement` appeared in all diagrams. Also, keep the style and connectors placement consistent. Maybe we can consider reducing the number of diagrams to favor maintainability

File: images/Decoratable, Classifier, and Property.png
Comment (claudenirmf) [inline images/Decoratable, Classifier, and Property.png:1]: - Maybe we should consider default value for the other attributes as well (e.g., `isDerived`), but these are heavily influenced by the stereotypes and nature restrictions.

File: images/ElementView.png
Comment (claudenirmf) [inline images/ElementView.png:1]: - Let's consider how ternary relations will affect the abstract syntax. Right now, `RelationView` cannot accommodate this need.
- I am not sure that "diagram owner" is an OntoUML thing. This seems like a Visual Paradigm thing =/

File: images/Classifier.png
Comment (claudenirmf) [inline images/Classifier.png:1]: - If we keep this view, it might make sense to add property stereotypes to it

File: images/Examples Class Diagram.png
Comment (claudenirmf) [inline images/Examples Class Diagram.png:1]: Example removed - unused.

File: images/Metamodel Summary Diagram.png
Comment (claudenirmf) [inline images/Metamodel Summary Diagram.png:1]: This color coding does not make sense outside the catalog paper

**Review by claudenirmf — COMMENTED** [review]
Please keep track of the discussion points included for when we have a team meeting

File: images/Abstract Syntax.png
Comment (pedropaulofb) [reply images/Abstract Syntax.png:1]: > What is `{CP1}` in `Project`? The same goes for `Class`

They are constraints. We need to translate them to OCL. For now we have the following ones:

- `Project`: 
  - A Project cannot have a project relation to another Project. I.e., Projects are not in the domain list of the project relation.
- `Class`:
  - Every Class that is a type (i.e., has OntologicalNature with value 'type'), must have order equal or greater than two.
  - Every Class that is not a type (i.e., that has OntologicalNature != "type") has order = 1.
  - All classes with order one have their isPowertype attribute equal to false.

> We need to change the datatype for multilingual text as discussed in the meeting.

Done.

> Do not model `CollectiveClass` as a subtype of `Class`. Move the attribute to `Class` and make it optional.

Agreed. Implemented.

File: images/Project.png
Comment (pedropaulofb) [reply images/Project.png:1]: > Review diagrams as you make changes (e.g., check if the additional attributes of OntoumlElement appeared in all diagrams.

Done to `OntoumlElement` and to `Class`.

> Maybe we can consider reducing the number of diagrams to favor maintainability

I totally agree.

File: images/Decoratable, Classifier, and Property.png
Comment (pedropaulofb) [reply images/Decoratable, Classifier, and Property.png:1]: > * Maybe we should consider default value for the other attributes as well (e.g., `isDerived`), but these are heavily influenced by the stereotypes and nature restrictions.

For now, I added multiplicities to all attributes and set default values to all boolean ones.

I agree that some are influenced by stereotypes and nature restrictions. We can create constraints regarding that.

**Comment by pedropaulofb** [comment]
Currently, we have the following "1st Level" classes:

- AggregationKind
- Cardinality
- OntologicalNature
- OntoumlElement
- Point
- Stereotype

The classes AggregationKind, OntologicalNature, and Stereotype are enumerations classes (extensional classes, defined by the set of their instances). So for me they are fine like this.

However, shouldn't we include Point as a OntoumlElement/DiagramElement and Cardinality as a OntoumlElement/ModelElement? IMHO, they are elements of these types and I see no reason to do not do that.

**Review by claudenirmf — APPROVED** [review]

**Pull request merged**